File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/c/abelsrzz.c
#include <stdio.h>
#include <stdbool.h>

// No existe página oficial, por lo que adjunto la wikipedia
// https://es.wikipedia.org/wiki/C_(lenguaje_de_programaci%C3%B3n)

//Esto es un comentario de una sola línea

/*
Esto es un comentario multilínea.
*/

const char GLOBAL[] =  "Constante global";
int VARIABLE = 10;

int ENTERO = 10000;
short ENTERO_CORTO = 5;
long ENTERO_LARGO = 1010010101;
long long ENTERO_MUY_LARGO = 999999999999999;
float DECIMAL = 10.54;
double DECIMAL_PRECISO = 10.44553434234;
long double DECIMAL_MUY_PRECISO = 43.3421342342341234141234;
char CARACTER = 1;
signed char CARACTER_FIRMADO = 1;
unsigned char CARACTER_SIN_SIGNO = 1;
bool BOOLEANO = true;

int main() {
    
    const char LOCAL[] = "Constante local";

    printf("Esto es una constante global: %s\n", GLOBAL);
    printf("Esto es una constante local: %s\n", LOCAL);
    printf("Esto es una variable de tipo int: %d\n", VARIABLE);

    printf("Hola mundo!");

}

File: Roadmap/00 - SINTAXIS, VARIABLES, TIPOS DE DATOS Y HOLA MUNDO/c#/abelsrzz.cs
// Documentación oficial: https://learn.microsoft.com/es-es/dotnet/csharp/
/*
Esto es un comentario de múltiples líneas.
*/
// Esto es un comentario de una sola línea


int numeroEntero = 42;
long numeroLargo = 1234567890123456789L;
float numeroFlotante = 3.14f;
double numeroDoble = 2.71828;
decimal numeroDecimal = 123.456m;

bool esVerdadero = true;
bool esFalso = false;

char caracter = 'A';

string texto = "Hola, mundo!";

DateTime fechaHoraActual = DateTime.Now;

List<int> listaEnteros = new List<int>() { 1, 2, 3 };
Dictionary<string, int> diccionario = new Dictionary<string, int>();

object objetoGenerico = 42;
dynamic variableDinamica = "Hola, dinámico!";

Console.WriteLine("Número Entero: " + numeroEntero);
Console.WriteLine("Número Largo: " + numeroLargo);
Console.WriteLine("Número Flotante: " + numeroFlotante);
Console.WriteLine("Número Doble: " + numeroDoble);
Console.WriteLine("Número Decimal: " + numeroDecimal);
Console.WriteLine("Es Verdadero: " + esVerdadero);
Console.WriteLine("Carácter: " + caracter);
Console.WriteLine("Cadena de Texto: " + texto);
Console.WriteLine("Fecha y Hora Actual: " + fechaHoraActual);
Console.WriteLine("Lista de Enteros: " + string.Join(", ", listaEnteros));
Console.WriteLine("Objeto Genérico: " + objetoGenerico);
Console.WriteLine("Variable Dinámica: " + variableDinamica);

Console.WriteLine("Hola mundo!");
